sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -351,6 +352,17 @@ public <T> T findById(Object id, Class<T> domainType, PartitionKey partitionKey)
     public <T> T findById(String containerName, Object id, Class<T> domainType) {
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(domainType, "domainType should not be null");
+        CosmosEntityInformation<?, ?> cosmosEntityInformation = CosmosEntityInformation.getInstance(domainType);
+        String containerPartitionKey = cosmosEntityInformation.getPartitionKeyFieldName();
+        if ("id".equals(containerPartitionKey) && id != null) {
+            try {
+                return findById(id, domainType, new PartitionKey(CosmosUtils.getStringIDValue(id)));
+            } catch (CosmosAccessException e) {
+                LOGGER.warn(e.getMessage());
+                return null;
+            }
+        }
+        LOGGER.warn("The partitionKey is not id!! Consider using findById(ID id, PartitionKey partitionKey) instead. See https://aka.ms/PointReadsInSpring for more info.");
```

Comment:
Logging a warning is a breaking change - and doesn't really help - like discussed before if we want to raise visibility the right way would eb to deprecate the findById method altogether (to raise warning at compile time vs. runtime)

Reply:
findById() is a Spring method which we cannot deprecate, even though this is a breaking change I believe it is a good approach moving forward.

Reply:
We have already seen CRIs when elevating log noise level. In many production environments logging with INFO is enabled - warnings/errors even more frequently. Logging a new message for every single method invocation is adding overhead (and making logs much harder to reason about). So, it is a perf and behavioral regression. FindById could easily be executed hundreds or thousands of times per second. What is the value in logging this warning over-and-over again. Why not log a warning or error only once?

Reply:
@TheovanKraay - I am absolutely fine with the behavior change (it is an improvement - no regression I am seeing there) - except for the logging. Customers having a working app right now where their log files aren't swamped with warnings might get hundreds or thousands of pretty much identical log lines after just upgrading the sdk. What value doe we provide my logging this for every method invocation?

Reply:
I think warning is fine as long as you only log it once per process - basically keep some state whether this warning was logged or not - that  way you don't cause too much noise, but still have the warning level logs. Do you consider that acceptable?

Reply:
Sounds fine to me. Kushagra wants some other changes too, will work on that tomorrow.

Reply:
Actually I don’t know if it makes sense to log only once per process. My thinking is that, if an app uses findById frequently, logging only once might be missed and not even noticed, so we may as well not have logged it at all. In one sense the whole idea is that if they are doing a cross pk query the “noise” of it being logged again and again causes them to actually change their app. The noise in their logs is better than the effect of doing cross pk queries if they have large containers and never realise until raising a support case. If they have small containers and/or low throughput, then the logging noise won’t matter so much anyway (but it still might cause them to change their app to remove the noise, and they’ll still be making the app more efficient in the end). So actually I think we should leave as WARN. My assumption is that many customers will not have WARN level in prod environments, but they will have it in pre-prod, so they’ll see this immediately and repeatedly when they upgrade versions, and that’s what we want. But I’m unsure what’s typical. @kushagraThapar @trande4884 any thoughts on this?

Reply:
Logs are emitted at runtime - when it is already too late to warn a developer. When making changes that require customers to change their application to consume a new SDK version we loose their trust in any new update. And in this case we know for sure that many customers use the findById for months - even the CRI I talked about yesterday (initially 10 partitions for few hundred documents) ran that service happily for several months. Starting to log >1000 log lines per second will force them to make changes to the application. And IMO that isn't justified. Either break  it at compile time and document it as a breaking change or avoid breaking the customer (as in don't force them to change their app). Single log per process I can live with because it nudges the cx and maybe some pick-up on it to improve their experience - but at least it doesn't break.

Reply:
We also have evidence showing that cx are not happy with increasing noisy   warning logs - Annie had added a log recently when channels got closed under certain non-happy-path circumstances - which also led to this warning being emitted for timeouts - we had 2 CRIs for it. We had similar CRIs for certain warning in RNTBD stack on certain error conditions.

Reply:
@TheovanKraay  - I agree with @FabianMeiswinkel  here. Almost every single production application will have WARN level logs enabled by default. Also, this is a performance regression for sure, as customers are very very sensitive to what gets logged or not. Its better to only log it once per application if possible (you can probably keep a boolean flag to make sure this only gets logged once).